optuna/study/_multi_objective.py

```diff
@@ -69,6 +74,94 @@ def _get_pareto_front_trials(study: "optuna.study.Study") -> List[FrozenTrial]:
     return _get_pareto_front_trials_by_trials(study.trials, study.directions)
 
 
+def _fast_non_dominated_sort(
+    objective_values: np.ndarray,
+    *,
+    penalty: np.ndarray | None = None,
+    n_below: int | None = None,
+) -> np.ndarray:
+    if penalty is None:
+        ranks, _ = _calculate_nondomination_rank(objective_values, n_below=n_below)
+        return ranks
+
+    if len(penalty) != len(objective_values):
+        raise ValueError(
```

Comment:
Could you add the test case for this line?

Reply:
Thanks for your comment. I added a test.